src/analysis/Environments/Environments.ts

```diff
@@ -350,43 +336,30 @@ export const Environments: React.FC<EnvironmentsProps> = (props) => {
     if (!_.some({ appName: errorAppId }, newApps)) {
       setErrorAppId(undefined);
     }
-    if (!_.some({ appName: deleteAppId }, newApps)) {
-      setDeleteAppId(undefined);
+    if (deleteAppModal.isOpen && !_.some({ appName: deleteAppModal.args?.appName }, newApps)) {
+      deleteAppModal.close();
     }
   });
   const loadData = withErrorIgnoring(refreshData);
 
   const pauseComputeAndRefresh = Utils.withBusyState(setLoading, async (compute: DecoratedComputeResource) => {
-    const wrappedPauseCompute = withErrorReporting('Error pausing compute', () =>
-      Utils.cond(
-        [
-          isRuntime(compute) && isGoogleWorkspaceInfo(compute.workspace),
-          () =>
-            ajax()
-              .Runtimes.runtimeWrapper(compute as RuntimeWithWorkspace)
-              .stop(),
-        ],
-        [
-          isRuntime(compute),
-          () =>
-            ajax()
-              .Runtimes.runtimeWrapper(compute as RuntimeWithWorkspace)
-              .stop(),
-        ],
-        [
-          isApp(compute) && isGoogleWorkspaceInfo(compute.workspace),
-          // @ts-expect-error
-          () => ajax().Apps.app(compute.workspace.googleProject, compute.appName).pause(),
-        ],
-        [
-          Utils.DEFAULT,
-          () => {
-            console.error('Pause is not currently implemented for azure apps');
-            return Promise.resolve();
-          },
-        ]
-      )
-    );
+    const wrappedPauseCompute = withErrorReporting('Error pausing compute', async () => {
+      if (isRuntime(compute) && isGoogleWorkspaceInfo(compute.workspace)) {
+        return leoRuntimeData.stop(compute);
+      }
+      if (isRuntime(compute)) {
+        return leoRuntimeData.stop(compute);
+      }
+      if (isApp(compute)) {
+        const computeWorkspace = compute.workspace;
+        if (isGoogleWorkspaceInfo(computeWorkspace)) {
+          return leoAppData.pause(computeWorkspace.googleProject, compute.appName);
+        }
+      }
+      // default:
+      console.error('Pause is not currently implemented for azure apps');
+      return Promise.resolve();
```

Comment:
```suggestion
```

This was to satisfy TS with matching return types for all cases of `cond`. With `cond` removed, it's not necessary.

Reply:
good catch :)